drivers/spi/spi_pl022.c

```diff
@@ -482,8 +482,9 @@ static void spi_pl022_xfer(const struct device *dev)
 		SSP_READ_REG(SSP_DR(cfg->reg));
 
 	while (data->rx_count < chunk_len || data->tx_count < chunk_len) {
+		size_t tx_cnt = 0;
 		/* Fill up fifo with available TX data */
-		while (SSP_TX_FIFO_NOT_FULL(cfg->reg) && data->tx_count < chunk_len) {
+		while (SSP_TX_FIFO_NOT_FULL(cfg->reg) && data->tx_count < chunk_len && tx_cnt < 8) {
```

Comment:
It should be better to use SSP_FIFO_DEPTH instead of 8.

Reply:
Yes, that would be better. Didn't see that.